util/gnmi.go

```diff
@@ -134,6 +134,41 @@ func FindPathElemPrefix(paths []*gpb.Path) *gpb.Path {
 	}
 }
 
+// pathElemMatchesQuery returns true if the given concrete (no wildcard)
+// PathElem is a possible gNMI telemetry name match for the reference PathElem.
+// In particular, it matches wildcards for names and list keys, but *not* "...".
+// See https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-path-conventions.md
+func pathElemMatchesQuery(elem, refElem *gpb.PathElem) bool {
+	if elem == nil || refElem == nil {
+		return elem == nil && refElem == nil
+	}
+
+	if refElem.Name != "*" && elem.Name != refElem.Name {
+		return false
+	}
+
+	for k, ref := range refElem.Key {
+		if v, ok := elem.Key[k]; !ok || (ref != "*" && v != ref) {
+			return false
+		}
+	}
+	return true
+}
+
+// PathElemsMatchQuery returns true if the given concrete (no wildcard)
+// PathElem path slice is a possible gNMI telemetry path match for the
+// reference PathElem path slice.
+// In particular, it matches wildcards for names and list keys, but *not* "...".
+// See https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-path-conventions.md
+func PathElemsMatchQuery(elems, refElems []*gpb.PathElem) bool {
```

Comment:
Similar comment here as above about making the concrete wording clearer, and the argument names.